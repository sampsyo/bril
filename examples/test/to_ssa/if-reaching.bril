@main(cond: bool) {
.entry:
    # Can be reached through `.left`, so it should be within a phi in `.zexit`.
    a: int = const 47;

    # Can be reached through `.right`, so it should be within a phi in `.zexit`.
    b: int = const 42;
    br cond .left .right;
.left:
    b: int = const 1;
    c: int = const 5;
    jmp .zexit;
.right:
    a: int = const 2;
    c: int = const 10;
    jmp .zexit;
.zexit:
    d: int = sub a c;
    print d;
}
