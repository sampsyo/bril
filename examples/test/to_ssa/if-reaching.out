@main(cond: bool) {
.entry:
  a.0: int = const 47;
  b.0: int = const 42;
  br cond .left .right;
.left:
  b.1: int = const 1;
  c.0: int = const 5;
  jmp .zexit;
.right:
  a.1: int = const 2;
  c.1: int = const 10;
  jmp .zexit;
.zexit:
  c.2: int = phi c.0 c.1 .left .right;
  b.2: int = phi b.1 b.0 .left .right;
  a.2: int = phi a.0 a.1 .left .right;
  d.0: int = sub a.2 c.2;
  print d.0;
  ret;
}
